lib/addons/velero/index.ts

```diff
@@ -0,0 +1,239 @@
+            this.options = deepmerge(defaultProps, props);
+        }
+        else {
+            this.options = defaultProps
+        }
+    }
+
+    deploy(clusterInfo: ClusterInfo): void {
+        const cluster = clusterInfo.cluster;
+        const props = this.options;
+        
+        let bucketName: string; // AWS S3 Bucketname
+        let veleroNamespace: string; // K8s namespace that Velero get deployed onto
+       
+        // Create S3 bucket if no existing bucket, create s3 bucket and corresponding KMS key
+        if ( !props.values.configuration.backupStorageLocation.bucket ){
+             console.log("existing S3 Bucket does not exists, creating S3 bucket");
+             const bucket = new s3.Bucket(cluster, 'velero-backup-bucket', {
+                encryption: s3.BucketEncryption.KMS_MANAGED, // Velero Known bug for support with S3 with SSE-KMS with CMK, thus it does not support S3 Bucket Key: https://github.com/vmware-tanzu/helm-charts/issues/83
+                blockPublicAccess: s3.BlockPublicAccess.BLOCK_ALL, // Block Public Access for S3
+                publicReadAccess: false
+            });
+
+            // S3 Bucket Policy for SSL Access
+            bucket.addToResourcePolicy(
+                new iam.PolicyStatement({
+                    sid: 'DenyHTTPTraffic',
+                    effect: iam.Effect.DENY,
+                    actions: ['s3:*'],
+                    resources: [bucket.arnForObjects('*')],
+                    principals: [new iam.AnyPrincipal()],
+                    conditions:{
+                        Bool:{
+                            'aws:SecureTransport': 'false',
+                        }
+                    }
+                })
+            );            
+            // Create S3 VPC Endpoint for the Velero pod to access S3 via VPC Endpoint instead of going to internet
+            cluster.vpc.addGatewayEndpoint('velero-backup-bucket-vpcEndPoint', {
+                service: GatewayVpcEndpointAwsService.S3
+            })
+            bucketName = bucket.bucketName
+        }
+        else {
+            bucketName = props.values.configuration.backupStorageLocation.bucket
+        }
+
+        // Create Namespace if not specified
+        if (props.namespace){
+            // Create Namespace if the 'create' option is false
+            if (props.namespace.create) {
+                console.log ("namespace:" + props.namespace.name + " does not existed, creating")
+                cluster.addManifest('velero-namespace',
+                {
+                    apiVersion: 'v1',
+                    kind: 'Namespace',
+                    metadata: { 
+                        name: props.namespace.name
+                    }
+                }
+                )
+                veleroNamespace = props.namespace.name;
+            }
+            else{
+                console.log ("namespace:" + props.namespace.name + " exists, not creating new namespace")
```

Comment:
We haven't been adding such console messages so this would stand out. Something for us to think about if we want to keep this and make it consistent across the board or remove it.

Reply:
if it is decided not to keep it more than happy to delete them.

Reply:
Cross them out as suggested.